### PR TITLE
chore(test): set up integration testing

### LINKED ISSUE #460: Integration test suite
### Feature Proposal

An integration test suite that will run karma-webpack against karma and webpack. This should increase confidence when updating dependencies and give us the ability to more easily test against different permutations of configuration options.

I have already started on a small testing package that I hope to finish up soon.

### PR DESCRIPTION
**Pull request by codymikol**
This will ensure that karma-webpack
can run as expected within a live
karma server.

Fixes #460

This PR contains a:

- [ ] **bugfix**
- [ ] new **feature**
- [ ] **code refactor**
- [x] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

This will make it easier to update dependencies as it will ensure that karma-webpack will still work in at least a basic setup. This also lays groundwork for adding additional integration tests as well for more niche scenarios.

### Breaking Changes

N/A

### Additional Info

This also introduces puppeteer as a dev dependency as I found it to be the most best headless option for doing an integration test.

### CODE REVIEW

File: test/integration/scenarios/basic-setup/basic-setup.test.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  const config = {
+    frameworks: ['webpack', 'mocha'],
+    files: [{ pattern: TEST_PATH }],
+    preprocessors: { [TEST_PATH]: ['webpack'] },
+    webpack: {},
+    browsers: ['ChromeHeadless'],
+    // Explicitly turn off reporters so the simulated test results are not confused with the actual results.
+    reporters: [],
+    port: 2389,
+    logLevel: 'OFF',
```
Comment (devoto13) [inline test/integration/scenarios/basic-setup/basic-setup.test.js:19]: Can this be the reason, why you don't see logs?

Comment (codymikol) [reply test/integration/scenarios/basic-setup/basic-setup.test.js:19]: This was added right at the end to keep the tests from being noisy, I'll try to get a repro up sometime today.

Comment (codymikol) [reply test/integration/scenarios/basic-setup/basic-setup.test.js:19]: Looking at this made me realize that I totally missed pushing one of my files though 🤦‍♂️

**Review by ryanclark — APPROVED** [review]
Awesome stuff

**Pull request merged**